src/TConsole.cpp

```diff
@@ -772,14 +777,31 @@ void TConsole::closeEvent( QCloseEvent *event )
 
     if( profile_name != "default_host" && ! mUserAgreedToCloseConsole )
     {
-        ASK: int choice = QMessageBox::question( this, "Exiting Session: Question", "Do you want to save the profile "+profile_name, QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel );
+        QCheckBox *cb = new QCheckBox( tr("Save the Window Layout before closing this profile.") );
+        QMessageBox *closePrompt = new QMessageBox( QMessageBox::Icon::Question, tr("Exiting Session: Question"), tr("Do you want to save the profile: %1").arg(profile_name), QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel, this );
+        closePrompt->setCheckBox(cb);
+
+        connect(cb, &QCheckBox::stateChanged, [this](int state){
+                if (static_cast<Qt::CheckState>(state) == Qt::CheckState::Checked) {
+                    this->mSaveLayoutRequested = true;
+                }
+            });
+
+        //ASK: int choice = QMessageBox::question( this, "Exiting Session: Question", "Do you want to save the profile "+profile_name, QMessageBox::Yes|QMessageBox::No|QMessageBox::Cancel );
+        ASK: mSaveLayoutRequested = false;
+        closePrompt->exec();
+        int choice = closePrompt->result();
```

Comment:
Yea this I was just unsure about.   
I have learned similarly that `goto` is to be shunned and avoided but this doesn't seem to be all too bad, its not a huge or confusing jump and it was of course easier for me to change less.  :stuck_out_tongue: